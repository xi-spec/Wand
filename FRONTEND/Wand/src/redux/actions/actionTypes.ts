export default {
  LOAD_PRODUCT_TYPE: 'LOAD_PRODUCT_TYPE',
  LOAD_ALL_PRODUCTS: 'LOAD_ALL_PRODUCTS',
  LOAD_NEW_PRODUCTS: 'LOAD_NEW_PRODUCTS',
  LOAD_PRODUCTS_LIST: 'LOAD_PRODUCTS_LIST',
  LOAD_PRODUCT_DETAIL: 'LOAD_PRODUCT_DETAIL',
  ADD_PRODUCT_TO_CART: ' ADD_PRODUCT_TO_CART',
  INCREASE_QUANTITY_CART: 'INCREASE_QUANTITY_CART',
  DECREASE_QUANTITY_CART: 'DECREASE_QUANTITY_CART',
  DELETE_PRODUCT_CART: 'DELETE_PRODUCT_CART',
  DELETE_ALL_CART: 'DELETE_ALL_CART',
  UPDATE_STOCK_API: 'UPDATE_STOCK_API',
  SEARCH_PRODUCT: 'SEARCH_PRODUCT',
  ADD_PRODUCT_TO_FAVOURITE: 'ADD_PRODUCT_TO_FAVOURITE',
  REGISTER: 'REGISTER',
  REGISTER_ERROR: 'REGISTER_ERROR',
  LOGIN: 'LOGIN',
  LOGIN_ERROR: 'LOGIN_ERROR',
  CLEAN_ISEXIST_STATUS: 'CLEAN_STATUS',
  CLEAN_LOGIN_ERROR: 'CLEAN_LOGIN_ERROR',
  CLEAN_REGISTED: 'CLEAN_REGISTED',
  REMOVE_FAVOURITE: 'REMOVE_FAVOURITE',
  LOGOUT: 'LOGOUT',
  UPDATE_USER: 'UPDATE_USER',
  PAYMENT_SUCCSSFULY: 'PAYMENT_SUCCESSFULY',
  PDATE_STOCK_API: 'PDATE_STOCK_API'
}
;
