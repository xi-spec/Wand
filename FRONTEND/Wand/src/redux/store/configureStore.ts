import reduxImmutableStatInvariant from 'redux-immutable-state-invariant';
import thunk from 'redux-thunk';
import { applyMiddleware, compose, createStore } from 'redux';
import rootReducer from '../reducers';

function configureStore () {
  const composeEnhancers = compose;

  return createStore(
    rootReducer,
    composeEnhancers(applyMiddleware(reduxImmutableStatInvariant(), thunk))
  );
}

export default configureStore();
