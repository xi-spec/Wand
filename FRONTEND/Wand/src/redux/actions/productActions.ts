import actionTypes from './actionTypes';
import axios from 'axios';
import SERVER from '../../constants/server';
import { Dispatch } from 'redux';
import { Product, ShoppingCart } from '../../models/initiaState';

export function loadProductsType () {
  return async (dispatch : Dispatch) => {
    const { data } = await axios.get(SERVER.URL_TYPE);
    dispatch({
      type: actionTypes.LOAD_PRODUCT_TYPE,
      data
    });
  };
}

export function loadAllProducts () {
  return async (dispatch : Dispatch) => {
    const { data } = await axios.get(SERVER.URL_PRODUCT);
    dispatch({
      type: actionTypes.LOAD_ALL_PRODUCTS,
      data
    });
  };
}

export function loadProductsList (typeName:string) {
  return {
    type: actionTypes.LOAD_PRODUCTS_LIST,
    typeName
  };
}

export function loadProductDetail (productId:string) {
  return {
    type: actionTypes.LOAD_PRODUCT_DETAIL,
    productId
  };
}

export function addProductToCart (product:Product) {
  return {
    type: actionTypes.ADD_PRODUCT_TO_CART,
    product
  };
}

export function increaseQuantityCart (productId:string) {
  return {
    type: actionTypes.INCREASE_QUANTITY_CART,
    productId
  };
}

export function decreaseQuantityCart (productId:string) {
  return {
    type: actionTypes.DECREASE_QUANTITY_CART,
    productId
  };
}

export function deleteProductCart (productId:string, quantity:number) {
  return {
    type: actionTypes.DELETE_PRODUCT_CART,
    productId,
    quantity
  };
}

export function searchProducts (name:string) {
  return {
    type: actionTypes.SEARCH_PRODUCT,
    name
  };
}

export function updateStockApi (cart:ShoppingCart) {
  return (dispatch : Dispatch) => {
    cart.cartList.forEach(async (product) => {
      const { data } = await axios.put(`${SERVER.URL_PRODUCT}/${product._id}`, { stock: product.stock });
      dispatch({
        type: actionTypes.UPDATE_STOCK_API,
        data
      });
    });
  };
}
