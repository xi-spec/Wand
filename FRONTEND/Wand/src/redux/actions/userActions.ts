import axios from 'axios';
import actionTypes from './actionTypes';
import SERVER from '../../constants/server';
import { User, Product, ShoppingCart } from '../../models/initiaState';
import { Dispatch } from 'redux';

export function addProductToFavourite (product:Product, user:User, favouriteState:boolean) {
  let newUser;
  favouriteState
    ? newUser = { ...user, favourite: [...user.favourite, product] }
    : newUser = { ...user, favourite: user.favourite.filter(elem => elem._id !== product._id) };

  return async (dispatch:Dispatch) => {
    const { data } = await axios.put(`${SERVER.URL_USER}/${user._id}`, newUser);
    dispatch({
      type: actionTypes.ADD_PRODUCT_TO_FAVOURITE,
      data
    });
  };
}

export function register (email:string, name:string, password:string) {
  const query = { email, name, password };
  let actionObj;
  return async (dispatch:Dispatch) => {
    try {
      const { data, status } = await axios.post(SERVER.URL_REGISTER, query);
      actionObj = {
        type: actionTypes.REGISTER,
        data,
        status
      };
    } catch (error) {
      actionObj = {
        type: actionTypes.REGISTER_ERROR
      };
    }
    dispatch(
      actionObj
    );
  };
}

export function login (email:string, password:string) {
  const query = { email, password };

  return async (dispatch:Dispatch) => {
    let actionObj;
    try {
      const { data } = await axios.post(SERVER.URL_LOGIN, query);
      actionObj = {
        type: actionTypes.LOGIN,
        data

      };
    } catch (error) {
      actionObj = {
        type: actionTypes.LOGIN_ERROR
      };
    }

    dispatch(actionObj);
  };
}

export function logout (email:string, password:string) {
  const query = { email, password };

  return async (dispatch:Dispatch) => {
    const { data } = await axios.post(SERVER.URL_LOGOUT, query);

    dispatch({
      type: actionTypes.LOGOUT,
      data
    });
  };
}

export function cleanIsExistStatus () {
  return {
    type: actionTypes.CLEAN_ISEXIST_STATUS
  };
}

export function cleanLoginError () {
  return {
    type: actionTypes.CLEAN_LOGIN_ERROR
  };
}

export function cleanRegisted () {
  return {
    type: actionTypes.CLEAN_REGISTED
  };
}

export function removeFavourite (product:Product, user:User) {
  const newUser = { ...user, favourite: user.favourite.filter(elem => elem._id !== product._id) };

  return async (dispatch:Dispatch) => {
    const { data } = await axios.put(`${SERVER.URL_USER}/${user._id}`, newUser);
    dispatch({
      type: actionTypes.REMOVE_FAVOURITE,
      data
    });
  };
}

export function updateAddress (street:string,
  city:string, country:string, postcode:string, user:User) {
  const query = { street, city, country, postcode };
  return async (dispatch:Dispatch) => {
    const { data } = await axios.put(`${SERVER.URL_USER}/${user._id}`, { address: query });
    dispatch({
      type: actionTypes.UPDATE_USER,
      data
    });
  };
}

export function updateCard (cardNumber:string, securtyCode:string, cardholder:string, expiryDate:string, user:User) {
  const query = { cardNumber, securtyCode, cardholder, expiryDate };
  return async (dispatch:Dispatch) => {
    const { data } = await axios.put(`${SERVER.URL_USER}/${user._id}`, { card: query });
    dispatch({
      type: actionTypes.UPDATE_USER,
      data
    });
  };
}

export function paymentSuccessfuly (user:User, cart:ShoppingCart) {
  return async (dispatch:Dispatch) => {
    const { data } = await axios.put(`${SERVER.URL_USER}/${user._id}`, { orderHistory: [...user.orderHistory, cart] });
    dispatch({
      type: actionTypes.PAYMENT_SUCCSSFULY,
      data
    });
  };
}
