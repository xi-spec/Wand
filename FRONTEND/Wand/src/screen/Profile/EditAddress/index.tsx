import React, { useState } from 'react';
import { Text, View, TouchableOpacity, ScrollView } from 'react-native';
import { User } from '../../../models/initiaState';
import styles from './styles';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { updateAddress } from '../../../redux/actions/userActions';
import { Input } from 'react-native-elements';

function EditAddress ({ user, navigation, actions }:
  {user:User, navigation:{goBack:Function}, actions:any}) {
  const { street, country, city, postcode } = user.address || { undefined };
  const [userStreet, setStreet] = useState(street);
  const [userCountry, setCountry] = useState(country);
  const [userCity, setCity] = useState(city);
  const [userPostcode, setPostcode] = useState(postcode);

  return (
           <ScrollView
           contentContainerStyle={styles.container}
    keyboardShouldPersistTaps='handled'>
         <View>

         <Input
         containerStyle={styles.input}
         inputStyle={{ width: '100%' }}
         label="Street"
         placeholder="street"
         value={userStreet}
         onChangeText={(text) => setStreet(text)}
         autoCompleteType='street-address'>
         </Input>

         <Input
         containerStyle={styles.input}
        value={userCity}
        onChangeText={(text) => setCity(text)}
        label="City"
         placeholder='city'>
         </Input>

         <Input
         containerStyle={styles.input}
        value={userCountry}
        onChangeText={(text) => setCountry(text)}
        label="Country"
         placeholder='country'>
         </Input>

         <Input
          containerStyle={styles.input}
        value={userPostcode}
        onChangeText={(text) => setPostcode(text)}
        label="Zip/Postcode"
         placeholder='zip/postcode'
         autoCompleteType='postal-code'
         keyboardType='numeric'>
         </Input>
         </View>

         <View style={styles.applyContainer}>
          <TouchableOpacity
          testID="apply"
          onPress={() => { actions.updateAddress(userStreet, userCity, userCountry, userPostcode, user); navigation.goBack(); }}
          disabled={!userStreet || !userCountry || !userCity || !userPostcode}
           style={styles.apply}>
            <Text style={styles.applyText}>APPLY</Text>
          </TouchableOpacity>
        </View>

    </ScrollView>
  );
}

function mapStateToProps ({ user }) {
  return {
    user
  };
}

function mapDispatchToProps (dispatch) {
  return {
    actions: bindActionCreators({
      updateAddress
    }, dispatch)
  };
}
export default connect(mapStateToProps, mapDispatchToProps)(EditAddress);
