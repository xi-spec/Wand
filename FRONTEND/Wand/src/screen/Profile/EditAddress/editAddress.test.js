import React from 'react';
import { render, fireEvent, act } from '@testing-library/react-native';
import configureStore from 'redux-mock-store';
import * as actions from '../../../redux/actions/userActions';
import { Provider } from 'react-redux';
import EditAddress from './index';

describe('Given a EditAddress component', () => {
  const mockStore = configureStore();
  let store;
  let navigation;

  beforeEach(() => {
    jest.spyOn(actions, 'updateAddress').mockReturnValueOnce({ type: '' });
    store = mockStore({
      user: {
        address: {
          street: '',
          country: '',
          city: '',
          postcode: ''
        }
      }
    });
    navigation = {
      goBack: jest.fn()
    };
  });
  describe('When is render', () => {
    test('Then is mathced to snapshot', () => {
      const tree = render(<Provider store={store}><EditAddress navigation={navigation} /></Provider>);

      expect(tree).toMatchSnapshot();
    });

    test('Then is mathced to snapshot without address', () => {
      store = mockStore({
        user: {

        }
      });
      const tree = render(<Provider store={store}><EditAddress navigation={navigation} /></Provider>);

      expect(tree).toMatchSnapshot();
    });
  });

  describe('When is typing in input', () => {
    test('Then is input value is change', () => {
      const { getByPlaceholderText } = render(<Provider store={store}><EditAddress navigation={navigation} /></Provider>);

      const streetInput = getByPlaceholderText('street');
      const countryInput = getByPlaceholderText('country');
      const zipInput = getByPlaceholderText('zip/postcode');
      const cityInput = getByPlaceholderText('city');
      const text = ('hola');

      act(() => {
        fireEvent.changeText(streetInput, text);
        fireEvent.changeText(countryInput, text);
        fireEvent.changeText(zipInput, text);
        fireEvent.changeText(cityInput, text);
      });
      expect(streetInput.props.value).toBe(text);
      expect(countryInput.props.value).toBe(text);
      expect(zipInput.props.value).toBe(text);
      expect(cityInput.props.value).toBe(text);
    });
  });

  describe('When apply button is pressed', () => {
    test('Then actions.updateAddressis invoked', () => {
      store = mockStore({
        user: {
          address: {
            street: 'hello',
            country: 'hello',
            city: 'hello',
            postcode: 'hello'
          }
        }
      });

      const { getByTestId } = render(<Provider store={store}><EditAddress navigation={navigation}/></Provider>);

      fireEvent.press(getByTestId('apply'));
      expect(actions.updateAddress).toHaveBeenCalled();
    });
  });
});
