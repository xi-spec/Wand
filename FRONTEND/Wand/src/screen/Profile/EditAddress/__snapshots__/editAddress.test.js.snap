// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a EditAddress component When is render Then is mathced to snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "flex": 1,
      "margin": 20,
    }
  }
  keyboardShouldPersistTaps="handled"
>
  <View>
    <View>
      <View
        style={
          Object {
            "borderRadius": 5,
            "height": 40,
            "marginBottom": 50,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": 350,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Street
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCompleteType="street-address"
            editable={true}
            onChangeText={[Function]}
            placeholder="street"
            placeholderTextColor="#86939e"
            rejectResponderTermination={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
                "width": "100%",
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "borderRadius": 5,
            "height": 40,
            "marginBottom": 50,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": 350,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          City
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            onChangeText={[Function]}
            placeholder="city"
            placeholderTextColor="#86939e"
            rejectResponderTermination={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "borderRadius": 5,
            "height": 40,
            "marginBottom": 50,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": 350,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Country
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            onChangeText={[Function]}
            placeholder="country"
            placeholderTextColor="#86939e"
            rejectResponderTermination={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "borderRadius": 5,
            "height": 40,
            "marginBottom": 50,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": 350,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Zip/Postcode
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCompleteType="postal-code"
            editable={true}
            keyboardType="numeric"
            onChangeText={[Function]}
            placeholder="zip/postcode"
            placeholderTextColor="#86939e"
            rejectResponderTermination={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "bottom": 20,
          "height": 30,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2E3192",
            "borderRadius": 5,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="apply"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "RalewayRegular",
              "fontSize": 18,
            }
          }
        >
          APPLY
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Given a EditAddress component When is render Then is mathced to snapshot without address 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "flex": 1,
      "margin": 20,
    }
  }
  keyboardShouldPersistTaps="handled"
>
  <View>
    <View>
      <View
        style={
          Object {
            "borderRadius": 5,
            "height": 40,
            "marginBottom": 50,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": 350,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Street
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCompleteType="street-address"
            editable={true}
            onChangeText={[Function]}
            placeholder="street"
            placeholderTextColor="#86939e"
            rejectResponderTermination={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
                "width": "100%",
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "borderRadius": 5,
            "height": 40,
            "marginBottom": 50,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": 350,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          City
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            onChangeText={[Function]}
            placeholder="city"
            placeholderTextColor="#86939e"
            rejectResponderTermination={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "borderRadius": 5,
            "height": 40,
            "marginBottom": 50,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": 350,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Country
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            onChangeText={[Function]}
            placeholder="country"
            placeholderTextColor="#86939e"
            rejectResponderTermination={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "borderRadius": 5,
            "height": 40,
            "marginBottom": 50,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": 350,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Zip/Postcode
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCompleteType="postal-code"
            editable={true}
            keyboardType="numeric"
            onChangeText={[Function]}
            placeholder="zip/postcode"
            placeholderTextColor="#86939e"
            rejectResponderTermination={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "bottom": 20,
          "height": 30,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2E3192",
            "borderRadius": 5,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="apply"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "RalewayRegular",
              "fontSize": 18,
            }
          }
        >
          APPLY
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
